xbmc/music/MusicThumbLoader.cpp

```diff
@@ -56,8 +56,7 @@ bool CMusicThumbLoader::LoadItemCached(CFileItem* pItem)
   if (pItem->m_bIsShareOrDrive)
     return false;
 
-  if (pItem->HasMusicInfoTag() && (pItem->GetArt().empty() ||
-    (pItem->GetArt().size() == 1 && pItem->HasArt("thumb"))))
+  if (pItem->HasMusicInfoTag() && !pItem->HasArt("thumb"))
```

Comment:
This fix is correct. FillLibraryArt needs to be called when either  there is no art (other than the default icon) or the only art was a "thumb"  (or a default icon ) so that thumbnails for all the types of art available for that item are loaded.  Just checking that there is no "thumb" is insufficient, we may have a "thumb" but need the other art types fetching.

Reply:
In effect `FillLibraryArt` needs to be called when it has not been called before. Maybe it is better to have a property that flags that, rather than check art empty contents except for "thumb" or "icon" (both art types that can be set via other routes).